File: code/cs/OpcReadNode.Server.cs
// DOC
var machineIsRunningNode = new OpcDataVariableNode<bool>("IsRunning", false);

// DOC
machineIsRunningNode.Value = true;

// DOC
machineIsRunningNode.Description = "My description";

// DOC
machineIsRunningNode.ApplyChanges(server.SystemContext);

// DOC
machineIsRunningNode.ReadDescriptionCallback = HandleReadDescription;
...
private OpcAttributeValue<string> HandleReadDescription(
        OpcReadAttributeValueContext context,
        OpcAttributeValue<string> value)
{
    return ReadDescriptionFromDataSource(context.Node) ?? value;
}

// DOC
machineIsRunningNode.ReadVariableValueCallback = HandleReadVariableValue;
...
private OpcVariableValue<object> HandleReadVariableValue(
        OpcReadVariableValueContext context,
        OpcVariableValue<object> value)
{
    return ReadValueFromDataSource(context.Node) ?? value;
}

File: code/vb/OpcReadNode.Server.vb
'DOC
Dim machineIsRunningNode = New OpcDataVariableNode(Of Boolean)("IsRunning", False)

'DOC
machineIsRunningNode.Value = True

'DOC
machineIsRunningNode.Description = "My description"

'DOC
machineIsRunningNode.ApplyChanges(server.SystemContext)

'DOC
machineIsRunningNode.ReadDescriptionCallback = AddressOf HandleReadDescription
...
Private Function HandleReadDescription( _
        ByVal context As OpcReadAttributeValueContext, _
        ByVal value As OpcAttributeValue(Of String)) As OpcAttributeValue(Of String)
    Return If(ReadDescriptionFromDataSource(context.Node), value)
End Function


'DOC
machineIsRunningNode.ReadVariableValueCallback = AddressOf HandleReadVariableValue
...
Private Function HandleReadVariableValue( _
        ByVal context As OpcReadVariableValueContext, _
        ByVal value As OpcVariableValue(Of Object)) As OpcVariableValue(Of Object)
    Return If(ReadValueFromDataSource(context.Node), value)
End Function

